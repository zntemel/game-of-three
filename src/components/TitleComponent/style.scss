.title-component {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  align-content: center;
  width: 80%;
  .ant-page-header-heading-title {
    font-size: 20px;
    font-weight: 400;
  }
}
