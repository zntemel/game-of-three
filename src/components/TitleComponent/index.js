import React from "react";
import PropTypes from "prop-types";
import { PageHeader } from "antd";

import "./style.scss";

function TitleComponent(props) {
  const { title, history } = props;

  return (
    <>
      <div className="title-component">
        <PageHeader
          className="game-page-header"
          onBack={() => history.push("/")}
          title={title}
        />
      </div>
    </>
  );
}

TitleComponent.propTypes = {
  title: PropTypes.string,
  history: PropTypes.object,
};

export default TitleComponent;
