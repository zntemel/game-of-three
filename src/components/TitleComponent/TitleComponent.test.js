import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import TitleComponent from "./index";

describe("TitleComponent", () => {
  const wrapper = shallow(<TitleComponent title={"sample title"} history={history} />);
  const instance = wrapper.instance();

  describe("shallow wrapper", () => {
    it("should render succesfully with snapshots", () => {
      const component = renderer.create(wrapper).toJSON();
      expect(component).toMatchSnapshot();
    });

    it("instance should be null because of the dumy component", () => {
      expect(instance).toEqual(null);
    });

    it("should render with correct title", () => {
      expect(wrapper.find("PageHeader").props().title).toEqual("sample title");
    });
  });
});
