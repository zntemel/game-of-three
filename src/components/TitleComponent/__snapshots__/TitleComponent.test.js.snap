// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleComponent shallow wrapper should render succesfully with snapshots 1`] = `
<div
  className="title-component"
>
  <div
    className="ant-page-header game-page-header ant-page-header-ghost"
  >
    <div
      className="ant-page-header-heading"
    >
      <div
        className="ant-page-header-heading-left"
      >
        <div
          className="ant-page-header-back"
        >
          <div
            aria-label="Back"
            className="ant-page-header-back-button"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            role="button"
            style={
              Object {
                "background": "transparent",
                "border": 0,
                "display": "inline-block",
                "lineHeight": "inherit",
                "padding": 0,
              }
            }
            tabIndex={0}
          >
            <span
              aria-label="arrow-left"
              className="anticon anticon-arrow-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="arrow-left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
          </div>
        </div>
        <span
          className="ant-page-header-heading-title"
          title="sample title"
        >
          sample title
        </span>
      </div>
    </div>
  </div>
</div>
`;
